homeassistant/components/melnor/manifest.json

```diff
@@ -1,9 +1,18 @@
 {
-  "domain": "melnor",
-  "name": "Melnor Bluetooth",
+  "after_dependencies": ["bluetooth"],
+  "bluetooth": [
+    {
+      "local_name": "YM*"
+    },
+    {
+      "manufacturer_data_start": [89]
+    }
```

Comment:
These devices are really odd in how little is advertised, and Bluez has issues parsing the `ADV_IND` portion of an advertisement packet. This portion of the packet happens to be used by the melnor devices for the rich manufacturer data. 

This means on Linux filtering on both name and rich advertising data essentially doesn't work at all. It works _sometimes_, but it's nondeterministic and there's nothing that can be done in any areas that we have any control over.

You won't reproduce this in a dev env on Mac or Windows, as those BT stacks are more robust and more forgiving.  

To work around this we'll cast a wider net here and look for _either_ the `local_name` _or_ the hexidecimal-ish (in bytes as integers) representation of `59*` which is the first 2 digits of every model number that we can support.